internal/transport/proxy_test.go

```diff
@@ -157,33 +181,62 @@ func testHTTPConnect(t *testing.T, proxyURLModify func(*url.URL) *url.URL, proxy
 	if string(recvBuf) != string(msg) {
 		t.Fatalf("received msg: %v, want %v", recvBuf, msg)
 	}
+
+	if len(args.serverMessage) > 0 {
+		c.SetReadDeadline(time.Now().Add(defaultTestTimeout))
```

Comment:
There's now 3 places where we're setting this deadline.

Maybe instead when the conn is created in `proxyDial` we should just do `c.SetDeadline(....defaultTestTimeout)` (note `Deadline` not `Read` to also abort writes)?

Reply:
During testing, I noticed that the test can hang in `proxyDial` if the proxy server waits for a `server hello` when the server doesn't intend to send one. This was due to a logical error on my part which I fixed.

To be safe, I added read deadlines in both the places where reads could hang.

> Maybe instead when the conn is created in proxyDial we should just do c.SetDeadline(....defaultTestTimeout) (note Deadline not Read to also abort writes)?

Do you mean adding a deadline in the actual `proxyDial()` implementation by introducing a timeout that only the tests use?
Or are you referring to adding a deadline on the conn returned by `proxyDial` in the test code?

The l latter would not prevent `proxyDial` from hanging when the proxy server is stuck reading. The former would work but it involves changing non-test code, which I was trying to avoid.

Reply:
Sorry, I didn't mean _in_ `proxyDial`... I meant the connection it returns, we could immediately, always, do a `SetDeadline(...)`, instead of doing it later and conditionally.

Reply:
Changed `SetReadDeadline` to `SetDeadline`. Moved the SetDeadline immediately after the `proxyDial` call. The `SetDeadline` in `proxyServer.run()` is still present to avoid proxyDial from hanging indefinitely in case of test failures.